### DISCUSSION TITLE
Mini.pick match items in an orderless style

### DISCUSSION THREAD

**Discussion by luathn** [nvim-mini/mini.nvim#1407] [Ideas]
Hi @echasnovski,
I use mini.pick and have a question. I hope you can answer it.
Do you have any plans to make mini.pick match items in any order, like fzf-lua does?

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/2fc8c88b-8245-4560-909b-8b918048e72a" />

Now I use mini.pick as the default picker in my config, so I hope this feature will be added to mini.pick.

**Comment by echasnovski** [comment]
> Do you have any plans to make mini.pick match items in any order, like fzf-lua does?

In theory this is possible via some additional flag on how ["grouped fuzzy"](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-pick.txt#L892-L893) mode operates. Now it indeed [matches in order](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-pick.txt#L868-L870) which is aligned with how overall fuzzy matching is done.

In practice, I don't think this is a good idea to begin with. All default matching in 'mini.pick' is done in query order and is a big factor when it comes to optimizing matching speed. So I don't think this will be implemented as built-in.

My best suggestion would be to implement this manually as [default `config.source.match`](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-pick.txt#L753) (it should follow [this specification](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-pick.txt#L308)).
